src/java/org/apache/cassandra/db/Mutation.java

```diff
@@ -237,7 +242,11 @@ public static Mutation merge(List<Mutation> mutations)
     public CompletableFuture<?> applyFuture(WriteOptions writeOptions)
     {
         Keyspace ks = Keyspace.open(keyspaceName);
-        return ks.applyFuture(this, writeOptions, true);
+        return ks.applyFuture(this, writeOptions, true).thenRun(() -> {
+            RequestSensors sensors = requestTracker.get();
+            if (sensors != null)
+                sensors.syncAllSensors();
```

Comment:
So `applyFuture` is the one called in response to `MutationVerbHandler`. The other `apply` methods seems to fit into one of those categories:
* Unit tests
* Virtual table (not durable)
* Read repairs
* Batch log replay
* Storing batch logs themselves as mutations

I think from the above we only need to handle read repairs (because in the happy case, we would've accounted for all replicas already so we should we for repaired ones - no double counting here)

For batch logs it seems the usage of mutations/replay is implementation detail and don;t fit in the same category as WRITE_BYTES for the effective mutation

Unites test no need, virtual tables we don't support in CNDB (and no bytes will be written to disk IIUC).

I probably missed something here but WDYT?

Reply:
One possibility is to filter in based on [WriteOptions](https://github.com/datastax/cassandra/blob/c21c71ccc7af18d33ddaf9164e2882abaf4b03ed/src/java/org/apache/cassandra/db/WriteOptions.java#L26)

We can start with something like:
```
Set<WriteOptions> TRACK_WRITE_BYTES_FOR = Set.of(DEFAULT, FOR_READ_REPAIR)  
```

And we `apply` sync on all `apply` paths, but filter based on `TRACK_WRITE_BYTES_FOR` - I'm just not 100% sure which writes option would qualify.

Reply:
> Am I making sense?

Yes, will update the `apply` code paths as such.